Sources/NIOHTTP2/StreamStateMachine.swift

```diff
@@ -1023,3 +1022,35 @@ private extension HPACKHeaders {
         return self.first { $0.name == ":status" }?.value.first == "1"
     }
 }
+
+
+/// State of a HTTP2 Stream.
+public enum NIOHTTP2StreamState {
```

Comment:
Can we elaborate on this doc string and explain what this type is for?

Reply:
I updated the doc string. How does it look now?

Reply:
Looks good. Given the type is now used only internally, can we make it `internal`?

Reply:
Will do, but from the original issue it's not clear to me if they would want access to the error from outside the library. I understand the wish to keep types internal or private unless there is an explicit need to be public.

Reply:
I grant that, but right now they can't actually _see_ the error from outside the library because the error that is thrown holds the stream state in an `internal` property. For now I think leaving things internal is fine: we can always make them public later.